src/components/admin/form-components/withSearchBar.jsx

```diff
@@ -0,0 +1,223 @@
+import React from 'react';
+import BaseComponent from 'lib/BaseComponent';
+import { capFirstLetter, slugify, debounce } from 'lib/utilities';
+import _ from 'lodash';
+
+import { updateFieldValue } from 'components/admin/lib/form-field-updaters';
+import { cleanupFalcorKeys } from 'lib/falcor/falcor-utilities';
+
+// material-ui
+import TextField from 'material-ui/TextField';
+import Menu from 'material-ui/Menu';
+import MenuItem from 'material-ui/MenuItem';
+import ContentAdd from 'material-ui/svg-icons/content/add';
+
+const withSearchBar = config => {
+  if (!('formatter' in config) || !_.isFunction(config.formatter)) {
+    throw new Error(
+      'withSearchBar.config.formatter must be supplied as a function.',
+    );
+  }
+  if (!('fields' in config) || !_.isArray(config.fields)) {
+    throw new Error(
+      'withSearchBar.config.fields must be supplied as an array.',
+    );
+  }
+  if (!('displayName' in config) || !_.isString(config.displayName)) {
+    throw new Error(
+      'withSearchBar.config.displayName must be supplied as a string.',
+    );
+  }
+  if (!('category' in config) || !_.isString(config.category)) {
+    throw new Error(
+      'withSearchBar.config.category must be supplied as a string.',
+    );
+  }
+
+  class SearchBar extends BaseComponent {
+    constructor(props) {
+      super(props);
+      this.fieldUpdaters = {
+        searchValue: updateFieldValue.bind(this, 'searchValue', undefined),
+      };
+      this.safeSetState({
+        searchValue: '',
+        searchSuggestions: [],
+      });
+
+      this.debouncedGetSuggestions = debounce(() => {
+        const query = this.state.searchValue;
+        if (!query.trim()) {
+          this.safeSetState({ searchSuggestions: [] });
+          return;
+        }
+
+        let pathSets = this.props.extraPathSets || [];
+        pathSets = pathSets.reduce(
+          (arr, pathSet) =>
+            arr.push(
+              ['search', 'posts', query, { length: this.props.length }].concat(
```

Comment:
This is a bug. The field is now called `articles`, but also this query wouldn't work for authors I am pretty sure.

Reply:
Good catch!